test/language/statements/class/elements/class-field-is-observable-by-proxy.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2019 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Public class fields initialization calls [[DefineOwnProperty]]
+esid: sec-define-field
+info: |
+  DefineField(receiver, fieldRecord)
+    ...
+    8. If fieldName is a Private Name,
+      a. Perform ? PrivateFieldAdd(fieldName, receiver, initValue).
+    9. Else,
+      a. Assert: IsPropertyKey(fieldName) is true.
+      b. Perform ? CreateDataPropertyOrThrow(receiver, fieldName, initValue).
+    10. Return.
+includes: [compareArray.js]
+features: [class, class-fields-public, Proxy]
+---*/
+
+let arr = [];
+
+class ProxyBase {
+  constructor() {
+    return new Proxy(this, {
+      defineProperty: function (target, key, descriptor) {
+        arr.push(key);
+        assert(descriptor.enumerable);
+        assert(descriptor.configurable);
+        assert(descriptor.writable);
```

Comment:
nit: in the future it's better to be explicit about the expected value being `true`, otherwise `assert()` will only expect for any true'ish vale.

also, good catch here verifying the descriptor created for [the data property](https://tc39.github.io/ecma262/#sec-createdataproperty)